File: packages/android_intent_plus/lib/src/put_classes/put_byte.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

/// Adds a byte to the bundle in the two's complement binary representation
/// of the specified [value], which must fit in a single byte.
///
/// In other words, [value] must be between -128 and 127, inclusive.
class PutByte extends PutBase<int> {
  PutByte({required String key, required int value})
      : super(key: key, value: value) {
    if (value < -128 || value > 127) {
      throw RangeError.value(
        value,
        "value must be between -128 and 127, inclusive.",
      );
    }
  }

  @override
  String get javaClass => 'PutByte';

  factory PutByte.fromJson({required String key, required dynamic value}) {
    return PutByte(key: key, value: value as int);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutByte.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutByte extends PutBase<Byte> {

  public static final String JAVA_CLASS = "PutByte";

  public PutByte(String key, Byte value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutByte putByte) {
    bundle.putByte(putByte.key, putByte.value);
  }

  public static PutByte fromJson(JSONObject jsonObject) throws JSONException {
    return new PutByte(
        jsonObject.getString(Constants.KEY), (byte) jsonObject.getInt(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", (int) value);
    return jsonObject;
  }
}
